apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2225,18 +2291,38 @@ static void libc_stdio_nulloutstream_tc(void)
 */
 static void libc_stdio_rawinstream_tc(void)
 {
-	int fd;
+	int fd, ret;
 	char *filename = VFS_FILE_PATH;
+	char *buf = VFS_TEST_CONTENTS_1;
 
 	struct lib_rawinstream_s rawinstream;
 
-	fd = open(filename, O_RDONLY);
+	/* Negative case, no character is read from the rawinstream */
+
+	fd = open(filename, O_WRONLY);
 	TC_ASSERT_GEQ("open", fd, 0);
 
 	lib_rawinstream((FAR struct lib_rawinstream_s *)&rawinstream, fd);
+	rawinstream.public.get(&rawinstream.public);
+	TC_ASSERT_EQ_CLEANUP("rawinstream_getc", rawinstream.public.nget, 0, close(fd));
+	TC_ASSERT_EQ_CLEANUP("lib_rawinstream", rawinstream.fd, fd, close(fd));
+
+	close(fd);
```

Comment:
As we said before, let's remove duplicated codes.
```
rawinstream.public.get(&rawinstream.public);
close(fd);
TC_ASSERT_EQ("rawinstream_getc", rawinstream.public.nget, 0);
TC_ASSERT_EQ("lib_rawinstream", rawinstream.fd, fd);
```

Reply:
checking for 
TC_ASSERT_EQ("lib_rawinstream", rawinstream.fd, fd);
after closing fd does not seem correct. 
Hence made it this way

Reply:
I though it as you did. But even though we call the close function, fd value is not changed I heard. Is your opinion that your codes looks right? We need to discuss this more. As per our decission, we need to modify many places.

Reply:
This approach looks cleaner & logical. 
Let me know wat we need to go for ?
1. Readability
2. Reducing code size

Reply:
@thapav As you said, It can looks correct logically that closing fd after checking it.
We think it is good to keep this and to update similar codes which were implemented before later.